### PR TITLE
Added code needed for cache warming

### PR DESCRIPTION
**Pull request by avasquez614**
Ticket craftercms/craftercms#3472

### CODE REVIEW

File: src/main/java/org/craftercms/core/service/Content.java [hunk 1]
```diff
@@ -39,6 +39,6 @@
     /**
      * Returns a {@link InputStream} for the data.
      */
-    public InputStream getInputStream() throws IOException;
+    InputStream getInputStream() throws IOException;
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/core/service/Content.java]: Why remove `public`?

File: src/main/java/org/craftercms/core/store/impl/AbstractCachedContentStoreAdapter.java [hunk 2]
```diff
@@ -124,16 +124,15 @@ public String toString() {
...
-        StoreException {
+    public List<Item> findItems(final Context context, final CachingOptions cachingOptions, final String path)
+            throws InvalidContextException, XmlFileParseException, StoreException {
         final CachingOptions actualCachingOptions = cachingOptions != null? cachingOptions: defaultCachingOptions;
 
         return cacheTemplate.getObject(context, actualCachingOptions, new Callback<List<Item>>() {
 
             @Override
             public List<Item> execute() {
-                List<Item> items = doFindItems(context, actualCachingOptions, path, withDescriptor);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/core/store/impl/AbstractCachedContentStoreAdapter.java]: What's `withDescriptor`?

File: src/main/java/org/craftercms/core/store/impl/filesystem/FileSystemContent.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.craftercms.core.store.impl.filesystem;
+
+import org.craftercms.core.service.Content;
+
+import java.io.*;
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/core/store/impl/filesystem/FileSystemContent.java]: Why not `nio`?

File: src/main/java/org/craftercms/core/util/ContentStoreUtils.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * Normalize the path: this means, append a leading '/' at the beginning and remove any trailing '/' (unless
+     * the path is root). This is done for consistency in handling paths.
+     *
+     * @param path
+     * @return the normalized path (with a leading '/' and without a trailing '/')
+     */
+    public static final String normalizePath(String path) {
+        if (!path.startsWith("/")) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/core/util/ContentStoreUtils.java]: I know we're on *nix, but why not use FS separator for correctness?

File: src/test/java/org/craftercms/core/store/impl/filesystem/FileSystemContentStoreAdapterTest.java [hunk 5]
```diff
@@ -129,24 +122,30 @@ public void testGetStaticAssetItem() throws Exception {
...
 
         Collections.sort(items, ItemComparator.INSTANCE);
 
         assertEquals(2, items.size());
-        assertCrafterCMSLogoItem(items.get(0));
-        assertDescriptorItem(items.get(1));
+
+        assertNotNull(items.get(0));
+
+        assertEquals(CRAFTER_CMS_LOGIC_NAME, items.get(0).getName());
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/core/store/impl/filesystem/FileSystemContentStoreAdapterTest.java]: What's a LOGIC_NAME?

File: src/main/java/org/craftercms/core/service/Content.java (hunk 1 above)
Comment (avasquez614) [reply src/main/java/org/craftercms/core/service/Content.java]: You don't need to specify `public` in interfaces, all methods are public

File: src/main/java/org/craftercms/core/util/ContentStoreUtils.java (hunk 4 above)
Comment (avasquez614) [reply src/main/java/org/craftercms/core/util/ContentStoreUtils.java]: These are actually URL paths, `/` will always be used.

File: src/main/java/org/craftercms/core/store/impl/filesystem/FileSystemContent.java (hunk 3 above)
Comment (avasquez614) [reply src/main/java/org/craftercms/core/store/impl/filesystem/FileSystemContent.java]: We can change it to `nio`, I just didn't want to change most of the io code that was already there.

File: src/test/java/org/craftercms/core/store/impl/filesystem/FileSystemContentStoreAdapterTest.java (hunk 5 above)
Comment (avasquez614) [reply src/test/java/org/craftercms/core/store/impl/filesystem/FileSystemContentStoreAdapterTest.java]: Typo it should be logo

**Pull request merged**